Source/memio.c

```diff
@@ -182,7 +182,7 @@ int mem_fseek(MEMFILE *stream, signed long position, mem_rel_t whence)
 			return -1;
 	}
 
-	if (newpos < stream->buflen)
+	if (newpos <= stream->buflen)
```

Comment:
I'm not sure about this change. Otherwise, MEM_SEEK_END never works, so there is definitely a bug in memio.c

Reply:
I was just going to ask about this change. Needs more investigation.

Reply:
Same fix in Odamex https://github.com/odamex/odamex/blob/7190b0217abfd7fb22ad6f9f10026324367e8f3a/common/m_memio.cpp#L200

Reply:
This must be right, else calling the function with `(..., 0, SEEK_END)` will always fail.

Reply:
Wait a minute. If `buflen == 1` then `newpos` should be the same as `stream->position`, else it would point exactly one byte past the end of the stream. Or is it supposed to work this way?

Reply:
I think it is supposed to work this way to match stdio behavior. Not sure I understand why `buflen == 1` case is different.

Reply:
Not different, just easier for me to understand as a case example. 😁

Reply:
Pointer difference must match buffer size. Right, sorry, forget my question. I was just thinking loudly.